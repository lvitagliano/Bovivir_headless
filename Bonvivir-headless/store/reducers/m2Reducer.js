import * as types from '../actionTypes'
import { normalizeError } from './utils/normalizer'

const initialState = {
  //Datos Personales
  step1: {
    firstname: '',
    lastname: '',
    gender: 1,
    email: '',
    areacod: '',
    days: '',
    months: '',
    years: '',
    is_subscribed: false,
  },
  //Datos de facturacion
  step2: {
    region: '',
    region_id: '',
    region_code: '',
    country_id: '',
    street: '',
    telephone: '',
    city: '',
    postcode: '',
    email: '',
    firstname: '',
    lastname: '',
    same_as_billing: '',
    save_in_address_book: '',
  },
  step3: {
    region: '',
    region_id: '',
    region_code: '',
    country_id: '',
    street: '',
    telephone: '',
    city: '',
    postcode: '',
    email: '',
    firstname: '',
    lastname: '',
    save_in_address_book: '',
  },
  step4: {
    paymentMethod: 'checkmo',
  },
  cart: {
    id: null,
    applied_coupons: [],
    items: [],
  },
  step5: {},
  customerData: {},
  customerWishList: {},
  customerBoughtList: {},
  paymentMethodData: {},
  paymentURL: '',
  currentStep: '',
  loading: true,
  formError: {},
  error: null,
}

const m2Reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case types.SET_M2_STEP:
      return set_m2_step(state, payload)
    case types.SET_REQUEST:
      return set_request(state, payload)
    case types.SET_ERROR:
      return set_error(state, payload)
    case types.SET_FORM_ERROR:
      return set_form_error(state, payload)
    case types.SET_CART_SUCCESS:
      return set_cart_success(state, payload)
    case types.SET_EMPTY_CART_SUCCESS:
      return set_empty_cart_success(state, payload)
    case types.SET_ITEM_SUCCESS:
      return set_item_success(state, payload)
    case types.DELETE_ITEM_SUCCESS:
      return delete_item_success(state, payload)
    case types.SET_CART_AND_ITEMS_SUCCESS:
      return set_cart_and_items_success(state, payload)
    case types.SET_CUSTOMER_DATA_SUCCESS:
      return set_customer_data_success(state, payload)
    case types.SET_CUSTOMER_WISHLIST_SUCCESS:
      return set_customer_wishlist_success(state, payload)
    case types.SET_CUSTOMER_ORDER_SUCCESS:
      return set_customer_order_success(state, payload)
    case types.SET_PAYMENT_METHOD_OPS_SUCCESS:
      return set_payment_method_ops_success(state, payload)
    case types.SET_PAYMENT_URL_SUCCESS:
      return set_payment_url_success(state, payload)

    case types.SET_SHIPPING_ADDRESS:
      return set_shipping_address(state, payload)
    case types.SET_BILLING_ADDRESS:
      return set_billing_address(state, payload)
    case types.SET_PAYMENT_METHOD:
      return set_payment_methods(state, payload)

    case types.SET_APPLIED_COUPONS:
      return set_applied_coupons(state, payload)
    case types.LOGOUT:
      return initialState
    default:
      return state
  }
}

const set_shipping_address = (state, data) => ({ ...state, step2: data })
const set_billing_address = (state, data) => ({ ...state, step3: data })
const set_payment_methods = (state, data) => ({ ...state, step4: data })

const set_m2_step = (state, data) => {
  const step = data.step
  const currentStep = data.currentStep
  delete data['step']
  delete data['currentStep']
  return {
    ...state,
    [step]: data,
    currentStep,
    loading: false,
    error: null,
  }
}

const set_request = (state, data) => ({ ...state, loading: true })

const set_error = (state, data) => ({ ...state, loading: false, error: normalizeError(data) })

const set_form_error = (state, data) => ({ ...state, formError: data })

const set_cart_success = (state, data) => {
  const clonedCart = { ...state.cart, quote_id: data }
  return { ...state, cart: clonedCart, loading: false, error: null }
}

const set_empty_cart_success = (state, data) => {
  return { ...state, cart: data, loading: false, error: null }
}

const set_item_success = (state, data) => {
  const clonedCart = { ...state.cart, items: [...state.cart.items, data] }
  return { ...state, cart: clonedCart, loading: false, error: null }
}

const delete_item_success = (state, data) => {
  const clonedCart = { ...state.cart, items: items.reduce(item => data.id !== item.id) }
  return { ...state, cart: clonedCart, loading: false, error: null }
}

const set_customer_data_success = (state, data) => {
  return { ...state, customerData: data, loading: false, error: null }
}

const set_customer_wishlist_success = (state, data) => {
  return { ...state, customerWishList: data, loading: false, error: null }
}

const set_customer_order_success = (state, data) => {
  return { ...state, customerBoughtList: data.customerOrders, loading: false, error: null }
}

const set_cart_and_items_success = (state, data) => {
  return { ...state, cart: data, loading: false, error: null }
}

const set_payment_method_ops_success = (state, data) => {
  return { ...state, paymentMethodData: data, loading: false, error: null }
}

const set_payment_url_success = (state, data) => {
  return { ...state, paymentURL: data[1], loading: false, error: null }
}

export { m2Reducer }
