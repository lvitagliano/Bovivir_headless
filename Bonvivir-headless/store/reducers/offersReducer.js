import * as types from '../actionTypes';

const initialState = {
    offers: [],
    offer:{},
    loading:true,
    error:null
}

export const offersReducer = (state=initialState,action) => {
    switch (action.type) {
        case types.GET_ALL_OFFERS:
            return {
                ...state,
                offers: action.payload,
                loading:false,
                error:null
            }
        default:
            return state
    }
}