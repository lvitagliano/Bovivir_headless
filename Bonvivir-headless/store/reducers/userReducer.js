import * as types from '../actionTypes'
import { normalizeError } from './utils/normalizer'

const initialState = {
  m2DataLogIn: '',
  loginRequested: false,
  auth0DataLogIn: {},
  isLogedInM2: false,
  isLogedInAuth0: false,
  isCreatedInM2: false,
  isCreatedInAuth0: false,
  error: {},
  loading: false,
}

export const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.SET_M2_LOGIN_SUCCESS:
      return set_m2_login_success(state, action.payload)
    case types.SET_AUTH0_LOGIN_SUCCESS:
      return set_auth0_login_success(state, action.payload)
    case types.SET_USER_REQUEST:
      return set_user_request(state, action.payload)
    case types.SET_USER_ERROR:
      return set_user_error(state, action.payload)
    case types.SET_LOGIN_REQUEST:
      return set_login_request(state, action.payload)
    case types.NEW_M2_SUCCESS:
      return new_m2_success(state)
    case types.NEW_AUTH0_SUCCESS:
      return new_auth0_success(state)
    case types.LOGOUT:
      return initialState
    default:
      return state
  }
}

const set_m2_login_success = (state, data) => ({ ...state, m2DataLogIn: data, isLogedInM2: true })

const set_auth0_login_success = (state, data) => ({
  ...state,
  auth0DataLogIn: data,
  isLogedInAuth0: true,
})

const set_login_request = (state, data) => ({ ...state, loginRequested: data })

const set_user_request = (state, data) => ({ ...state, loading: true })

const set_user_error = (state, data) => ({ ...state, loading: false, error: normalizeError(data) })

const new_m2_success = state => ({ ...state, isCreatedInM2: true })

const new_auth0_success = state => ({ ...state, isCreatedInAuth0: true })
