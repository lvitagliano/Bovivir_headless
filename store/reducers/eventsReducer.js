import * as types from '../actionTypes'

const initialState = {
    allEvents: {
        events: [],
        loading: false,
        error: null,
        loaded: false
    }
}

export const eventsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_ALL_EVENTS_START:
      return {
        ...state,
        allEvents: {
            ...state.allEvents,
            loading: true,

        }
      }
    case types.GET_ALL_EVENTS_SUCCESS:
      return {
        ...state,
        allEvents: {
            ...state.allEvents,
            events: action.payload,
            loading: false,

        }
      }
    case types.GET_ALL_EVENTS_FAIL:
      return {
        ...state,
        allEvents: {
          ...state.allEvents,
          error: action.error,
          loading: false,
        },
      }

    default:
      return state
  }
}
