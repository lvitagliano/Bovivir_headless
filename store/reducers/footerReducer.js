import * as types from '../actionTypes'

const initialState = {
    socialMedia: [],
}

export const footerReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_SOCIAL_MEDIA:
      return {
        ...state,
        socialMedia: action.payload
      }
       default:
        return state
    }
}



