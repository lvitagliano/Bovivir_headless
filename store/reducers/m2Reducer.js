import * as types from '../actionTypes'
import { normalizeError } from './utils/normalizer'

const initialState = {
  //Datos Personales
  step1: {
    firstname: '',
    lastname: '',
    gender: 1,
    email: '',
    areacod: '',
    days: '',
    months: '',
    years: '',
    is_subscribed: false,
  },
  //Datos de facturacion
  step2: {
    country_id: '',
    street: '',
    telephone: '',
    city: '',
    postcode: '',
    email: '',
    firstname: '',
    lastname: '',
    same_as_billing: '',
    save_in_address_book: '',
  },
  step3: {
    country_id: '',
    street: '',
    telephone: '',
    city: '',
    postcode: '',
    email: '',
    firstname: '',
    lastname: '',
    save_in_address_book: '',
  },
  step4: {
    paymentMethod: 'checkmo',
  },
  cart: {
    id: null,
    applied_coupons: [],
    items: [],
  },
  step5: {},
  addressView: {},
  customerData: {},
  customerWishList: {},
  customerBoughtList: {},
  customerAddresses: {},
  paymentMethodData: {},
  paymentURL: '',
  currentStep: '',
  loading: true,
  formError: {},
  error: null,
  success: null,
  pointHOP: [],
  selectedShipping: {},
  openTooWeak: false,
}

const m2Reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case types.SET_CLEAR_ALERTS:
      return set_clear_alerts(state, payload)
    case types.SET_M2_STEP:
      return set_m2_step(state, payload)
    case types.SET_REQUEST:
      return set_request(state, payload)
    case types.SET_ERROR:
      return set_error(state, payload)
    case types.SET_SUCCESS:
      return set_success(state, payload)
    case types.SET_FORM_ERROR:
      return set_form_error(state, payload)
    case types.SET_CART_SUCCESS:
      return set_cart_success(state, payload)
    case types.SET_EMPTY_CART_SUCCESS:
      return set_empty_cart_success(state, payload)
    case types.SET_ITEM_SUCCESS:
      return set_item_success(state, payload)
    case types.DELETE_ITEM_SUCCESS:
      return delete_item_success(state, payload)
    case types.SET_CART_AND_ITEMS_SUCCESS:
      return set_cart_and_items_success(state, payload)
    case types.SET_CUSTOMER_DATA_SUCCESS:
      return set_customer_data_success(state, payload)
    case types.SET_CUSTOMER_WISHLIST_SUCCESS:
      return set_customer_wishlist_success(state, payload)
    case types.SET_CUSTOMER_ORDER_SUCCESS:
      return set_customer_order_success(state, payload)
    case types.SET_PAYMENT_METHOD_OPS_SUCCESS:
      return set_payment_method_ops_success(state, payload)
    case types.SET_PAYMENT_URL_SUCCESS:
      return set_payment_url_success(state, payload)
    case types.SET_SHIPPING_ADDRESS:
      return set_shipping_address(state, payload)
    case types.SET_SHIPPING_VIEW_ADDRESS:
      return set_shipping_view_address(state, payload)
    case types.SET_BILLING_ADDRESS:
      return set_billing_address(state, payload)
    case types.SET_POINT_HOP:
      return set_shipping_methods(state, payload)
    case types.SET_PAYMENT_METHOD:
      return set_payment_methods(state, payload)
    case types.SET_APPLIED_COUPONS:
      return set_applied_coupons(state, payload)
    case types.SET_SELECTED_SHIPPING_METHOD:
      return set_selected_shipping_method(state, payload)
    case types.LOGOUT:
      return initialState
    case types.OPENTOOWEAK:
      return set_opentooweak(state, payload)
    default:
      return state
  }
}

const set_opentooweak = (state, data) => ({ ...state, openTooWeak: data })
const set_shipping_address = (state, data) => ({ ...state, step2: data })
const set_shipping_view_address = (state, data) => ({ ...state, addressView: data })
const set_billing_address = (state, data) => ({ ...state, step3: data })
const set_shipping_methods = (state, data) => ({ ...state, step5: data })
const set_payment_methods = (state, data) => ({ ...state, step4: data })
const set_selected_shipping_method = (state, data) => ({ ...state, selectedShipping: data })

const set_m2_step = (state, data) => {
  const step = data.step
  const currentStep = data.currentStep
  delete data['step']
  delete data['currentStep']
  return {
    ...state,
    currentStep,
    loading: false,
    error: null,
  }
}
const set_request = (state, data) => ({ ...state, loading: true })
const set_error = (state, data) => ({ ...state, loading: false, error: normalizeError(data) })
const set_success = (state, data) => ({ ...state, loading: false, success: normalizeError(data) })
const set_form_error = (state, data) => ({ ...state, formError: data })
const set_cart_success = (state, data) => {
  const clonedCart = { ...state.cart, quote_id: data }
  return { ...state, cart: clonedCart, loading: false, error: null }
}
const set_empty_cart_success = (state, data) => {
  return { ...state, cart: data, loading: false, error: null }
}
const set_item_success = (state, data) => {
  const clonedCart = { ...state.cart, items: [...state.cart.items, data] }
  return { ...state, cart: clonedCart, loading: false, error: null }
}
const delete_item_success = (state, data) => {
  const clonedCart = { ...state.cart, items: items.reduce(item => data.id !== item.id) }
  return { ...state, cart: clonedCart, loading: false, error: null }
}
const set_customer_data_success = (state, data) => {
  return { ...state, customerData: data, loading: false, error: null }
}
const set_customer_wishlist_success = (state, data) => {
  return { ...state, customerWishList: data, loading: false, error: null }
}
const set_customer_order_success = (state, data) => {
  return { ...state, customerBoughtList: data.customerOrders, loading: false, error: null }
}
const set_cart_and_items_success = (state, data) => {
  return { ...state, cart: data, loading: false, error: null }
}
const set_payment_method_ops_success = (state, data) => {
  return { ...state, paymentMethodData: data, loading: false, error: null }
}
const set_payment_url_success = (state, data) => {
  return { ...state, paymentURL: data[1], loading: false, error: null }
}

const set_clear_alerts = (state, data) => ({ ...state, error: null, success: null })

export { m2Reducer }
