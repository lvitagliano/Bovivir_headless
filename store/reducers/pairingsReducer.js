import * as types from '../actionTypes'

const initialState = {
  allPairings: {
    pairings: [],
    loading: false,
    error: null,
    loaded: false,
  },
}

export const pairingsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.GET_ALL_PAIRINGS_START:
      return {
        ...state,
        allPairings: {
          ...state.allPairings,
          loading: true,
        },
      }
    case types.GET_ALL_PAIRINGS_SUCCESS:
      return {
        ...state,
        allPairings: {
          ...state.allPairings,
          pairings: action.payload,
          loading: false,
        },
      }
    case types.GET_ALL_PAIRINGS_FAIL:
      return {
        ...state,
        allPairings: {
          ...state.allPairings,
          error: action.error,
          loading: false,
        },
      }

    default:
      return state
  }
}
