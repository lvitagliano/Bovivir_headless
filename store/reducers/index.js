import {combineReducers} from 'redux';
import { offersReducer } from './offersReducer';
import { registerReducer } from './registerReducer';
import { wordpressReducer } from './wordpressReducer';
import { m2Reducer } from './m2Reducer';
import {userReducer} from './userReducer';

export default combineReducers({
    offers: offersReducer,
    register: registerReducer,
    wordpress: wordpressReducer,
    m2: m2Reducer,
    user: userReducer
})
