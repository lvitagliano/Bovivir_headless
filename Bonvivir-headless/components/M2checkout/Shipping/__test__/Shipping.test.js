import React from 'react'
import { render, fireEvent, cleanup, waitFor, getByTestId } from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import { Shipping } from '../index'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import 'mutationobserver-shim'
import { m2Reducer } from '../../../../store/reducers/m2Reducer'
import { act } from 'react-dom/test-utils'

afterEach(cleanup)

const initState = { m2: m2Reducer(undefined, {}) }

function renderWithRedux(
  component,
  { initialState = initState, store = createStore(m2Reducer, initialState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return render(component, { wrapper: Wrapper, ...renderOptions })
}

describe('Shipping Data', () => {
  it('renders properly', () => {
    const wrapper = renderWithRedux(<Shipping />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('changes toggle', () => {
    const { getByRole, getByTestId, getByText, container } = renderWithRedux(<Shipping />)
    expect(getByRole('checkbox')).toHaveProperty('checked', true)
    getByRole('checkbox').click()
    fireEvent.change(getByRole('checkbox'), { target: { checked: false } })
    expect(getByRole('checkbox')).toHaveProperty('checked', false)
  })

  describe('valid inputs', () => {
    it('does not render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)

      await act(async () => {
        const firstName = container.querySelector('input[name="firstname"]')
        fireEvent.change(firstName, { target: { value: 'John' } })
        fireEvent.blur(firstName)
        const lastName = container.querySelector('input[name="lastname"]')
        fireEvent.change(lastName, { target: { value: 'Doe' } })
        fireEvent.blur(lastName)
        const telephone = container.querySelector('input[name="telephone"]')
        fireEvent.change(telephone, { target: { value: '11234456456' } })
        fireEvent.blur(telephone)
        const street = container.querySelector('input[name="street"]')
        fireEvent.change(street, { target: { value: 'Calle' } })
        fireEvent.blur(street)
        const number = container.querySelector('input[name="number"]')
        fireEvent.change(number, { target: { value: '123' } })
        fireEvent.blur(number)
        const postcode = container.querySelector('input[name="postcode"]')
        fireEvent.change(postcode, { target: { value: '3999' } })
        fireEvent.blur(postcode)
      })

      expect(container.innerHTML.includes('class="InputForm__Error')).toBe(false)
    })
  })

  describe('invalid firstname', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const firstName = container.querySelector('input[name="firstname"]')
      await act(async () => {
        fireEvent.change(firstName, { target: { value: '' } })
        fireEvent.blur(firstName)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(firstName, { target: { value: 'Prueba123' } })
        fireEvent.blur(firstName)
      })
      expect(container.innerHTML).toMatch('El campo solo debe contener letras')
    })

    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const firstName = container.querySelector('input[name="firstname"]')
      await act(async () => {
        fireEvent.change(firstName, { target: { value: 'aaaaaaaaaaaaaaaaaaaaa' } })
        fireEvent.blur(firstName)
      })
      expect(container.innerHTML).toMatch('El campo no debe tener mas de')
    })
  })

  describe('invalid lastname', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const lastname = container.querySelector('input[name="lastname"]')
      await act(async () => {
        fireEvent.change(lastname, { target: { value: '' } })
        fireEvent.blur(lastname)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(lastname, { target: { value: 'Prueba123' } })
        fireEvent.blur(lastname)
      })
      expect(container.innerHTML).toMatch('El campo solo debe contener letras')
    })

    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const lastname = container.querySelector('input[name="lastname"]')
      await act(async () => {
        fireEvent.change(lastname, { target: { value: 'aaaaaaaaaaaaaaaaaaaaa' } })
        fireEvent.blur(lastname)
      })
      expect(container.innerHTML).toMatch('El campo no debe tener mas de')
    })
  })

  describe('invalid telephone', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const telephone = container.querySelector('input[name="telephone"]')

      await act(async () => {
        fireEvent.change(telephone, { target: { value: '' } })
        fireEvent.blur(telephone)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(telephone, { target: { value: '123' } })
        fireEvent.blur(telephone)
      })
      expect(container.innerHTML).toMatch('El campo debe tener mas de')
    })

    it('does render validation errors 2', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const telephone = container.querySelector('input[name="telephone"]')

      await act(async () => {
        fireEvent.change(telephone, { target: { value: 'asdadasdas' } })
        fireEvent.blur(telephone)
      })
      expect(container.innerHTML).toMatch('El campo solo debe')

      await act(async () => {
        fireEvent.change(telephone, { target: { value: '1231231231233' } })
        fireEvent.blur(telephone)
      })
      expect(container.innerHTML).toMatch('El campo no debe tener mas de')
    })
  })

  describe('invalid street', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const street = container.querySelector('input[name="street"]')
      await act(async () => {
        fireEvent.change(street, { target: { value: '' } })
        fireEvent.blur(street)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(street, { target: { value: 'Prueba123' } })
        fireEvent.blur(street)
      })
      expect(container.innerHTML).toMatch('El campo solo debe contener letras')
    })

    it('does render validation errors 2', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const street = container.querySelector('input[name="street"]')
      await act(async () => {
        fireEvent.change(street, { target: { value: 'aaaaaaaaaaaaaaaaaaaaa' } })
        fireEvent.blur(street)
      })

      expect(container.innerHTML).toMatch('El campo no debe tener mas de')
    })
  })

  describe('invalid number', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const number = container.querySelector('input[name="number"]')
      await act(async () => {
        fireEvent.change(number, { target: { value: '' } })
        fireEvent.blur(number)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(number, { target: { value: 'P123' } })
        fireEvent.blur(number)
      })
      expect(container.innerHTML).toMatch('El campo solo debe contener números')
    })

    it('does render validation errors 2', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const number = container.querySelector('input[name="number"]')
      await act(async () => {
        fireEvent.change(number, { target: { value: '123456' } })
        fireEvent.blur(number)
      })

      expect(container.innerHTML).toMatch('El campo no debe tener mas de')
    })
  })

  describe('invalid postcode', () => {
    it('does render validation errors', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const postcode = container.querySelector('input[name="postcode"]')
      await act(async () => {
        fireEvent.change(postcode, { target: { value: '' } })
        fireEvent.blur(postcode)
      })
      expect(container.innerHTML).toMatch('Campo Requerido')

      await act(async () => {
        fireEvent.change(postcode, { target: { value: 'P123' } })
        fireEvent.blur(postcode)
      })
      expect(container.innerHTML).toMatch('El campo solo debe contener números')
    })

    it('does render validation errors 2', async () => {
      const { container } = renderWithRedux(<Shipping />)
      const postcode = container.querySelector('input[name="postcode"]')
      await act(async () => {
        fireEvent.change(postcode, { target: { value: '123456' } })
        fireEvent.blur(postcode)
      })

      expect(container.innerHTML).toMatch('El campo no debe tener mas de')

      await act(async () => {
        fireEvent.change(postcode, { target: { value: '1' } })
        fireEvent.blur(postcode)
      })

      expect(container.innerHTML).toMatch('El campo debe tener mas de')
    })
  })
})
