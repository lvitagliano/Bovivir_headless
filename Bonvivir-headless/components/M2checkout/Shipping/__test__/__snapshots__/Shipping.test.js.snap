// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shipping Data renders properly 1`] = `
<DocumentFragment>
  <form>
    <section
      class="styles__SectionDataContainer-sc-849pqq-0 EOzVa"
    >
      <div
        class="styles__FormCard-sc-849pqq-2 eocFuN"
        width="90%"
      >
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 knnnuq"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-1 MuiSwitch-colorSecondary PrivateSwitchBase-checked-17 Mui-checked WithStyles(ForwardRef(Switch))-checked-2"
              data-testid="toggle"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-19 MuiSwitch-input"
                  name="checkbox"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-3"
            />
          </span>
          <p>
            Enviar a esta dirección
          </p>
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 fSJYUJ"
            width="45%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Nombres *
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 cYuDKr"
              disabled=""
              name="firstname"
              placeholder="John"
              type="text"
              value=""
              width="100%"
            />
          </div>
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 fSJYUJ"
            width="45%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Apellidos *
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 cYuDKr"
              disabled=""
              name="lastname"
              placeholder="Doe"
              type="text"
              value=""
              width="100%"
            />
          </div>
        </div>
        <div
          class="styles__ContainerOneColumn-sc-849pqq-4 laNRzh"
          width="100%"
        >
          <label
            class="LabelForm-sc-1dbpd2j-0 hybBcM"
          >
            Número telefónico *
          </label>
          <input
            class="InputForm__InputFormBase-sc-55v2ww-0 cYuDKr"
            disabled=""
            name="telephone"
            placeholder="1112345678"
            type="text"
            value=""
            width="100%"
          />
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 blZfOP"
            width="65%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Calle *
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 boIKID"
              disabled=""
              name="street"
              placeholder="Avenida del Libertador"
              type="text"
              value=""
            />
          </div>
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 cxDNMr"
            width="30%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Altura *
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 boIKID"
              disabled=""
              name="number"
              placeholder="101"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 cxDNMr"
            width="30%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Código Postal *
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 boIKID"
              disabled=""
              name="postcode"
              placeholder="1638"
              type="text"
              value=""
            />
          </div>
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 cxDNMr"
            width="30%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Piso
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 boIKID"
              disabled=""
              name="floor"
              placeholder="10"
              type="text"
              value=""
            />
          </div>
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 cxDNMr"
            width="30%"
          >
            <label
              class="LabelForm-sc-1dbpd2j-0 hybBcM"
            >
              Depto.
            </label>
            <input
              class="InputForm__InputFormBase-sc-55v2ww-0 boIKID"
              disabled=""
              name="apartm"
              placeholder="203"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 fCNnjz"
            width="70%"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <input
          hidden=""
          name="regionData"
        />
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <div
            class="styles__ContainerOneColumn-sc-849pqq-4 blZfOP"
            width="65%"
          >
            <div
              class="MuiFormControl-root"
            >
              <b>
                Tipo de envio *
              </b>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                style="height: 32px;"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby=" "
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect"
                  role="button"
                  tabindex="0"
                >
                  Enviado
                </div>
                <input
                  type="hidden"
                  value="flatrate"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-106 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-107"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <p
            class="styles__ShipmentDescription-pg3cxi-0 fWZuzK"
          >
            Retiro en un punto de entrega, Solo AMBA: entre 48 a 72 hs habiles.
          </p>
        </div>
        <div
          class="styles__ContainerOneRow-sc-849pqq-5 hjtEVl"
        >
          <button
            class="styles__CustomButton-sc-849pqq-8 hQUwVG secondary"
            style="margin: 0px;"
            width="fit-content"
          >
            <p
              style="text-transform: uppercase; margin: 0px;"
            >
              calcular envio
            </p>
          </button>
          <b
            class="styles__ShipmentValue-pg3cxi-1 iMjKUG"
          />
        </div>
        <input
          hidden=""
          name="step"
          value=""
        />
        <input
          hidden=""
          name="id"
          value="0"
        />
      </div>
    </section>
    <input
      style="display: none;"
      type="submit"
    />
  </form>
</DocumentFragment>
`;
